src/rgw/rgw_sync_module_pubsub.cc

```diff
@@ -1217,6 +1217,7 @@ class RGWPSHandleObjEventCR : public RGWCoroutine {
             yield PSManager::call_get_subscription_cr(sync_env, env->manager, this, *oiter, *siter, &sub);
```

Comment:
this is failing here not because the subscription is missing (we are in a loop iterating over subscriptions of a topic), but because we try it twice (in user/global conf) - and it may fail on one of the cases.
if no subscription exist at all (at this point of the code) we should signal a failure. but this should happen in line 1244.

in addition, failures for storing the event or sending to an endpoint may happen in lines 1226 and 1234. if -ENOENT is not a good result code, and may be returned from them, we may want to change it to a different error code

Reply:
ok - i'm not familiar with the user/global distinction. but from the log, it looks like the first iteration was succeeding, and the second one failed here - and since this was the last error, it's what we returned up the stack with set_cr_error()

re: ENOENT, we have some special handling in RGWBucketSyncSingleEntryCR:
rgw_data_sync.cc:2713 `if (retcode < 0 && retcode != -ENOENT) {`

..so that if we try to sync an object that's no longer there, we just move on instead of retrying. we might want to move that logic into the sync modules where that makes sense, so that it doesn't apply to pubsub by default. but the only distinction is whether that error should be retried until it succeeds before further progress can be made in bucket sync

Reply:
> ok - i'm not familiar with the user/global distinction.
configuration could be global or at user level, so one of the fetching is expected to fail.
in line 1244 there is a check, that at-least one was ok. if both failed, it means there is something wrong, and the whole CR should fail.

re: ENOENT, i guess, that missing configuration for how to carry on with pubsub (which result in ENOENT at the pubsub level) should be translated to EINVAL at syn level.